tests/store/artifact/test_hdfs_artifact_repo.py

```diff
@@ -191,3 +191,18 @@ def test_download_artifacts():
                             os.path.join(tmp_dir.path(), artifact_path))
         with open(os.path.join(tmp_dir.path(), artifact_path), "rb") as fd:
             assert expected_data == fd.read()
+
+
+@mock.patch('pyarrow.hdfs.HadoopFileSystem', spec=HadoopFileSystem)
+def test_download_file(hdfs_system_mock):
+    expected_data = b"hello"
+    artifact_paths = ["hdfs://host/test.txt", "hdfs://host/test.txt/"]
+    hdfs_system_mock.return_value.open = mock_open(read_data=expected_data)
+
+    for artifact_path in artifact_paths:
+        hdfs_repo = HdfsArtifactRepository(artifact_path)
+        with TempDir() as tmp_dir:
+            hdfs_repo._download_file(artifact_path,
```

Comment:
use `download_artifacts` instead of `_download_file`.

in fact, we are missing a test that `_download_file` raises an error